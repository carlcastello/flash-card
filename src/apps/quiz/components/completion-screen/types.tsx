export interface IOwnProps {
  onClose: () => void,
  onReview: () => void,
  classes: {
    
  }
}