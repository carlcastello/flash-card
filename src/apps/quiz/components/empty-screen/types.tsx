export interface IOwnProps {
  onEdit: () => void,
  onClose: () => void,
  classes: {

  }
}