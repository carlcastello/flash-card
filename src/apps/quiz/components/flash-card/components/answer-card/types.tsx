import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  answer: string,
  flashcardStatus: FlashcardStatus,
  ref: any,
  classes: {
    paper: string,
    input: string,
    typography: string
  }
}

export interface IOwnState {
  answer: string,
  hideAnswer: boolean,
}