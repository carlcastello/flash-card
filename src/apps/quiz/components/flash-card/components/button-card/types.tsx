import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  flashcardStatus: FlashcardStatus,
  submit: () => void,
  next: () => void,
  classes: {
    button: string
  }
}

export interface IOwnState {

}