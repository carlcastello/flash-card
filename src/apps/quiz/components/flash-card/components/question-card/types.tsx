import { FlashCardStatus } from "../../types";
import { QuestionType } from "../../../../../commons/types";

export interface IQuestionObject {
  question: string,
  subQuestion?: string,
  hint?: string,
  questionType: QuestionType,
}

export interface IOwnProps {
  flashCardStatus: FlashCardStatus,
  questionObject: IQuestionObject,
  onSkip: () => void,
  classes: {
    paper: string,
    boxButtonContainer: string,
    boxQuestionaireContainer: string
  }
}