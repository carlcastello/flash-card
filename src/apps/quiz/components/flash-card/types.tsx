import { IFlashCard } from "../../../commons/types";

export enum FlashCardStatus {
  DEFAULT = "DEFAULT",
  CORRECT = "CORRECT",
  WRONG = "WRONG",
  HINT = "HINT"
}

export interface IOwnProps {
  update: (flashCardStatus: FlashCardStatus) => void,
  next: () => void,
  flashCardObject: IFlashCard,
  classes: {

  }
}

export interface IOwnState {
  flashCardStatus: FlashCardStatus
}