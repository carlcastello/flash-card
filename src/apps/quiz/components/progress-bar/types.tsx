import { FlashcardStatus } from "../flash-card/types";

export interface IOwnProps {
  currentQuestion: number,
  totalQuestion: number,
  flashcardStatus: FlashcardStatus,
  classes: {
    title: string,
    progressBarContainer: string,
    progressBar: string,
  }
}