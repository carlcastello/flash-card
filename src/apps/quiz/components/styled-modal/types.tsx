export interface IOwnProps {
  isOpen: boolean,
  onConfirm?: () => void,
  onIgnore: () => void,

  classes: {
    modalContainer: string,
    modalPaperContainer: string,
    closeButton: string,
  }
}