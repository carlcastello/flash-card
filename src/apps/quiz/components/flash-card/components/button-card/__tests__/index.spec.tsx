import React from "react";

import { ButtonCard } from "../index";
import { FlashcardStatus } from "../../../types";
import { shallow } from "enzyme";
import { Typography, Button } from "@material-ui/core";
import toJson from "enzyme-to-json";
 
describe("The Button Card", () => {

  let focusFunction: any;
  let blurFunction: any;
  let submitFunction: any;
  let nextFunction: any;

  let wrapper: any;
  let instance: any;

  beforeEach(() => {

    focusFunction = jest.fn();
    blurFunction = jest.fn();
    submitFunction = jest.fn();
    nextFunction = jest.fn();

    wrapper = shallow(
      <ButtonCard
        flashcardStatus={FlashcardStatus.DEFAULT}
        submit={submitFunction}
        next={nextFunction}
        classes={{
          button: "button"
        }}/>
    );

    instance = wrapper.instance();
    instance.buttonRef = { focus: focusFunction, blur: blurFunction };
  });

  describe("The render function",  () => {
    it("shows the submit button", () => {
      expect(wrapper.find(Typography).props().children).toEqual("Submit");
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("shows the next button", () => {
      wrapper.setProps({ flashcardStatus: FlashcardStatus.CORRECT });

      expect(wrapper.find(Typography).props().children).toEqual("Next");
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe("The componentDidUpdate", () => {
    it ("calls focusFunction if it updates to any status but flashcardStatus.DEFAULT", () => {
      expect(focusFunction).toHaveBeenCalledTimes(0);
      expect(blurFunction).toHaveBeenCalledTimes(0);

      wrapper.setProps({ flashcardStatus: FlashcardStatus.CORRECT });

      expect(focusFunction).toHaveBeenCalledTimes(1);
      expect(blurFunction).toHaveBeenCalledTimes(0);
    });

    it ("calls blurFunction if it updates to flashcardStatus.DEFAULT", () => {
      expect(focusFunction).toHaveBeenCalledTimes(0);
      expect(blurFunction).toHaveBeenCalledTimes(0);

      // Updating the props to trigger on blur between CORRECT -> DEFAULT transition
      wrapper.setProps({ flashcardStatus: FlashcardStatus.CORRECT });

      expect(focusFunction).toHaveBeenCalledTimes(1);
      expect(blurFunction).toHaveBeenCalledTimes(0);

      wrapper.setProps({ flashcardStatus: FlashcardStatus.DEFAULT });

      expect(focusFunction).toHaveBeenCalledTimes(1);
      expect(blurFunction).toHaveBeenCalledTimes(1);
    });
  });

  describe("The onClickFunction", () => {
    let preventDefaultFunction: any;

    beforeEach(() => {
      preventDefaultFunction = jest.fn();
    });

    it("calls the onSubmitFunction if FlashcardStatus.DEFAULT", () => {
      wrapper.find(Button).simulate("click", { preventDefault: preventDefaultFunction })

      expect(preventDefaultFunction).toHaveBeenCalled();
      expect(submitFunction).toHaveBeenCalled();
    });

    it("calls the onNextFunction", () => {
      wrapper.setProps({ flashcardStatus:  FlashcardStatus.WRONG });

      wrapper.find(Button).simulate("click", { preventDefault: preventDefaultFunction })

      expect(preventDefaultFunction).toHaveBeenCalled();
      expect(nextFunction).toHaveBeenCalled();
    });
  });
});