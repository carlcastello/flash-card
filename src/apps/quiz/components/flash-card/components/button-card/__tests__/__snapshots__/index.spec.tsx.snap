// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button Card The render function shows the next button 1`] = `
<Styled(MuiBox)
  pt={2.5}
>
  <WithStyles(ForwardRef(Paper))
    elevation={3}
  >
    <WithStyles(ForwardRef(Button))
      className="button CORRECT"
      onClick={[Function]}
      type="submit"
    >
      <WithStyles(ForwardRef(Typography))
        variant="h6"
      >
        Next
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Paper))>
</Styled(MuiBox)>
`;

exports[`The Button Card The render function shows the submit button 1`] = `
<Styled(MuiBox)
  pt={2.5}
>
  <WithStyles(ForwardRef(Paper))
    elevation={3}
  >
    <WithStyles(ForwardRef(Button))
      className="button DEFAULT"
      onClick={[Function]}
      type="submit"
    >
      <WithStyles(ForwardRef(Typography))
        variant="h6"
      >
        Submit
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Paper))>
</Styled(MuiBox)>
`;
